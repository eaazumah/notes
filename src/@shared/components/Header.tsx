import React from 'react';
import styled from 'styled-components/native';
import useTheme from '../hooks/useTheme';
import Icon from './Icon';

interface Props {
  title: string;
}

const Header = (props: Props) => {
  const {theme} = useTheme();
  return (
    <Container testID={'header'}>
      <Title testID={'title'}>{props.title}</Title>
      <Settings>
        <Icon
          size={25}
          name={'settings'}
          type={'Feather'}
          color={theme.primaryText}
        />
      </Settings>
    </Container>
  );
};

const Container = styled.View`
  width: 100%;
  height: 45px;
  margin-top: 50px;
  flex-direction: row;
  padding-horizontal: 17px;
  justify-content: space-between;
`;

const Title = styled.Text`
  font-size: 30px;
  font-weight: 800;
  color: ${props => props.theme.primaryText};
`;

const Settings = styled.TouchableOpacity``;

export default Header;
