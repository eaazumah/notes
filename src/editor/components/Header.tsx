import {useNavigation} from '@react-navigation/native';
import React from 'react';
import styled from 'styled-components/native';
import Icon from '../../@shared/components/Icon';
import useTheme from '../../@shared/hooks/useTheme';

interface Props {
  isArchived?: boolean;
  isFavorite?: boolean;
  toggleNoteArchived?: () => void;
  toggleNoteFavorite?: () => void;
}

const Header: React.FC<Props> = props => {
  const {theme} = useTheme();
  const navigation = useNavigation();
  return (
    <Container testID={'header-editor'}>
      <Left onPress={navigation.goBack}>
        <Icon name={'arrow-back'} color={theme.primaryText} />
      </Left>
      <Right>
        <Star onPress={props.toggleNoteFavorite}>
          <Icon
            size={28}
            type={'Feather'}
            name={'bookmark'}
            color={props.isFavorite ? theme.primary : theme.primaryText}
          />
        </Star>
        <Archive onPress={props.toggleNoteArchived}>
          <Icon
            size={28}
            type={'MaterialCommunityIcons'}
            name={'archive-arrow-down-outline'}
            color={props.isArchived ? theme.primary : theme.primaryText}
          />
        </Archive>
      </Right>
    </Container>
  );
};

const Container = styled.View`
  width: 100%;
  margin-top: 50px;
  flex-direction: row;
  padding-horizontal: 17px;
  justify-content: space-between;
`;

const Left = styled.TouchableOpacity``;

const Right = styled.View`
  flex-direction: row;
`;

const Star = styled.TouchableOpacity``;

const Archive = styled.TouchableOpacity``;

export default Header;
