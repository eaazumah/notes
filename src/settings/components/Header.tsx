import {useNavigation} from '@react-navigation/native';
import React from 'react';
import styled from 'styled-components/native';
import Icon from '../../@shared/components/Icon';
import useTheme from '../../@shared/hooks/useTheme';

interface Props {}

const Header: React.FC<Props> = () => {
  const {theme} = useTheme();
  const navigation = useNavigation();
  return (
    <Container testID={'header-editor'}>
      <Left onPress={navigation.goBack}>
        <Icon size={25} name={'arrow-back'} color={theme.primaryText} />
      </Left>
      <Right />
    </Container>
  );
};

const Container = styled.View`
  width: 100%;
  margin-top: 50px;
  flex-direction: row;
  padding-horizontal: 17px;
  justify-content: space-between;
`;

const Left = styled.TouchableOpacity``;

const Right = styled.View`
  flex-direction: row;
`;

export default Header;
