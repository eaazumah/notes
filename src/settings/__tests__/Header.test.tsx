jest.useFakeTimers();
import {render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import Header from '../components/Header';

describe('Settings Header', () => {
  test('Header renders correctly', () => {
    const screen = render(
      <TestProvider>
        <Header />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
