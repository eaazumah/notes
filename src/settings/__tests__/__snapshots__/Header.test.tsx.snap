// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Header Header renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 50,
        "paddingHorizontal": 17,
        "width": "100%",
      },
    ]
  }
  testID="header-editor"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#242A37"
      name="arrow-back"
      size={25}
      testID="icon"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  />
</View>
`;
